### ISSUE TITLE
[Bug]: [PartitionKey] should return error if setting a non existing  field as partition key field

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#24161]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230515-8b8f9f1e
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus-2.2.9.dev8
```

### Current Behavior

create the schema successfully if setting non existing field as partition key field

### Expected Behavior

return an error

**Comment by xige-16** [comment]
fixed on pymilvus

**Comment by yanliang567** [comment]
verified fix on pymilvus-2.2.9.dev11
